[Bug] Source file based metadata generation use wrong version of .NET ref assemblies

**Describe the bug**
When explicitly  specifying .NET SDK version by `global.json` with following settings.
Solution/Project build runs as expected with the specified SDK version.

But C# source file based builds always referencing the latest version of ref assemblies. 
(e.g. `C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\9.0.0-preview.1.24080.9`)

Because ref assemblies directory is resolved by version name of `Max()`.
https://github.com/dotnet/docfx/blob/841fcdfe3d97b958caeaf78b86c6d183f20f854f/src/Docfx.Dotnet/CompilationHelper.cs#L127-L129

**To Reproduce**
1. Use build environment that install .NET8/.NET9 SDKS.
2. Create following `global.json` file at docfx root directory.
```
{
  "sdk": {
    "version": "8.0.200",
    "rollforward" : "disable"
  }
}
```
3. Run `docfx metadata` command that targeting source file based build. 
4. Confirm following messages are displayed.

> Using .NET Core SDK 8.0.200
> Compiling C# files using .NET SDK 9.0.0-preview.1.24080.9 for net9.0

**Expected behavior**
Ref assemblies folder is resolved by using `Environment.Version`. (For temporary workaround)

It might be better to use `TargetFramework` version that is specified by docfx.json metadata properties.
https://dotnet.github.io/docfx/reference/docfx-json-reference.html#properties
And use `NETStandard.Library.Ref` DLLs when specified target to `netstandard2.1`.

**Context (please complete the following information):**
- OS: Windows
- Docfx version: [e.g. 2.75.3]

**Additional context**
This issue cause problems when adding support to reference external DLL features to 
[CreateCompilationFromCSharpFiles](https://github.com/dotnet/docfx/blob/841fcdfe3d97b958caeaf78b86c6d183f20f854f/src/Docfx.Dotnet/CompilationHelper.cs#L55-L62) method.

Currently following version mismatch warnings are occurred.
When adding additional DLL that target `.NET 8`. And  run `docfx metadata` commands

> warning: warning CS1701: Assuming assembly reference 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' used by 'R3' matches identity 'System.Runtime, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' of 'System.Runtime', you may need to supply runtime policy

--- a/src/Docfx.Dotnet/CompilationHelper.cs
+++ b/src/Docfx.Dotnet/CompilationHelper.cs
@@ -136,10 +136,14 @@ internal static class CompilationHelper
     {
         try
         {
+            // Get current .NET runtime version with `{major}.{minor}` format.
+            var dotnetMajorMinorVersion = Environment.Version.ToString(2);
+
+            // Resolve .NET SDK packs directory path. (e.g. `C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\8.0.2\ref\net8.0`)
             var dotnetExeDirectory = DotNetCorePathFinder.FindDotNetExeDirectory();
             var refDirectory = Path.Combine(dotnetExeDirectory, "packs/Microsoft.NETCore.App.Ref");
-            var version = new DirectoryInfo(refDirectory).GetDirectories().Select(d => d.Name).Max()!;
-            var moniker = new DirectoryInfo(Path.Combine(refDirectory, version, "ref")).GetDirectories().Select(d => d.Name).Max()!;
+            var version = new DirectoryInfo(refDirectory).GetDirectories().Select(d => d.Name).Where(x => x.StartsWith(dotnetMajorMinorVersion)).Max()!;
+            var moniker = new DirectoryInfo(Path.Combine(refDirectory, version, "ref")).GetDirectories().Select(d => d.Name).Where(x => x.EndsWith(dotnetMajorMinorVersion)).Max()!;
             var path = Path.Combine(refDirectory, version, "ref", moniker);
 
             Logger.LogInfo($"Compiling {language} files using .NET SDK {version} for {moniker}");

--- a/test/Docfx.Dotnet.Tests/GenerateMetadataFromCSUnitTest.cs
+++ b/test/Docfx.Dotnet.Tests/GenerateMetadataFromCSUnitTest.cs
@@ -2580,7 +2580,12 @@ namespace Test1
 
         var undefined = output.Items[0].Items[0].Items[1];
         Assert.NotNull(undefined);
+
+#if NET8_0_OR_GREATER
         Assert.Equal(@"public void Undefined(ConsoleKey key = ConsoleKey.None)", undefined.Syntax.Content[SyntaxLanguage.CSharp]);
+#else
+        Assert.Equal(@"public void Undefined(ConsoleKey key = default)", undefined.Syntax.Content[SyntaxLanguage.CSharp]);
+#endif
     }
 
     [Fact]
@@ -2648,8 +2653,12 @@ namespace Test1
 
         var enumUndefinedDefault = output.Items[0].Items[0].Items[3];
         Assert.NotNull(enumUndefinedDefault);
-        Assert.Equal(@"public void EnumUndefinedDefault(ConsoleKey? key = ConsoleKey.None)", enumUndefinedDefault.Syntax.Content[SyntaxLanguage.CSharp]);
 
+#if NET8_0_OR_GREATER
+        Assert.Equal(@"public void EnumUndefinedDefault(ConsoleKey? key = ConsoleKey.None)", enumUndefinedDefault.Syntax.Content[SyntaxLanguage.CSharp]);
+#else
+        Assert.Equal(@"public void EnumUndefinedDefault(ConsoleKey? key = null)", enumUndefinedDefault.Syntax.Content[SyntaxLanguage.CSharp]);
+#endif
         var enumUndefinedValue = output.Items[0].Items[0].Items[4];
         Assert.NotNull(enumUndefinedValue);
         Assert.Equal(@"public void EnumUndefinedValue(ConsoleKey? key = (ConsoleKey)999)", enumUndefinedValue.Syntax.Content[SyntaxLanguage.CSharp]);

